security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -323,18 +323,16 @@ func (sc *SecretManagerClient) addFileWatcher(file string, resourceName string)
 	if err := sc.tryAddFileWatcher(file, resourceName); err == nil {
 		return
 	}
-	// Retry file watcher as some times it might fail to add and we will miss change
+	// RetryWithContext file watcher as some times it might fail to add and we will miss change
 	// notifications on those files. For now, retry for ever till the watcher is added.
 	// TODO(ramaraochavali): Think about tieing these failures to liveness probe with a
 	// reasonable threshold (when the problem is not transient) and restart the pod.
 	go func() {
 		b := backoff.NewExponentialBackOff()
-		for {
-			if err := sc.tryAddFileWatcher(file, resourceName); err == nil {
-				break
-			}
-			time.Sleep(b.NextBackOff())
-		}
+		_ = b.RetryWithContext(context.TODO(), func() error {
```

Comment:
There are a lot of usages of `RetryWithContext` without a context.

I would recommend we make Retry and RetryContext (no With, to match stdlib usage)

Then this whole thing could be

```go
backoff.NewExponentialBackOff().Retry(func() error { return sc.tryAddFileWatcher(file, resourceName) })
```

Reply:
IMO, this will make things more complex. If you look at the k8s, previously the client-go does not have `Context` at all.  Just at a release, community introduced it.

Reply:
We should have one way to timeout though, not 2. If that's always context then that's fine with me but we should get rid of MaxElapsedTime

Reply:
yes, i removed the maxelapsedtime